### ISSUE TITLE
A question about bundle-adjusting

### ISSUE THREAD

**Issue by Wjt-shift** [chenhsuanlin/bundle-adjusting-NeRF#68]
Thanks for your excellent work!
I have a question about code.How to express bundle-adjustment in your code?I find you refine the pose with 6 parameters (R,t),using `torch.nn.embedding()`function ,setting Adam optimiter with adjustment learning rate,and modifying the position encoding function.But i can not find some options about computing jacobian matrix.Could you tell me the difference with nerf-- 's pose refine directly use network parameters?

**Comment by chenhsuanlin** [comment]
Hi @Wjt-shift, bundle adjustment, as a more general definition, is the process of optimizing the scene structure and the cameras altogether. So in the BARF context, the entire process is a type of bundle adjustment. The Jacobian matrices here are not explicitly constructed -- the (first-order) gradients are still formed via backprop, so this is different from classical bundle adjustment à la SfM/SLAM. BARF represents the pose refinement with learnable 6-DoF parameters, which (I believe) is the same as NeRF--; the high-level difference between NeRF-- and BARF is that (a) NeRF-- also optimizes the intrinsic camera parameters, and (b) NeRF-- does not have a coarse-to-fine optimization mechanism. Hope these help!

**Comment by Wjt-shift** [comment]
> 

thanks for your reply!thank you a lot!
